pkg/gcp/client/updater.go

```diff
@@ -192,16 +192,20 @@ func (u *updater) NAT(ctx context.Context, region string, router *compute.Router
 		return nat.Name == desired.Name
 	})
 
+	routerPatch := compute.Router{
+		Name: router.Name,
+	}
 	modified := false
 	// not found case
 	if index < 0 {
-		router.Nats = append(router.Nats, desired)
+		routerPatch.Nats = append(router.Nats, desired)
 		modified = true
 	} else {
 		nats := router.Nats[index]
 		if !reflect.DeepEqual(nats, desired) {
 			modified = true
 			router.Nats[index] = desired
```

Comment:
Is it ok to modify the original router object here?

Reply:
I will try to improve on that, but it doesn't make a difference. 

The router object is also replaced at the end of the patch call. In practice, the nats slice is modified in place due to a lack of deepCopy but since there is only one concurrent actor on the router at each time.